### PR TITLE
Feature / Add pinning to baltable and seperate out table components

### PR DESCRIPTION
**Pull request by 1saf**
# Description

Adds the ability to pin table data rows depending on data keys! Also implements that pinning feature into the veBAL table

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

VeBAL table should be pinned

## Visual context

image

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Review by garethfuller — CHANGES_REQUESTED** [review]
Nice work! BalTable is so much cleaner. There are a couple of linting issues to clean up.

File: src/components/contextual/pages/vebal/LMVoting/GaugesTable.vue [hunk 1]
```diff
@@ -167,13 +171,17 @@ function redirectToPool(gauge: VotingGaugeWithVotes) {
         sortColumn: 'nextPeriodVotes',
         sortDirection: 'desc'
       }"
+      :pin="{
+        pinOn: 'address',
+        pinnedData: ['0xE867AD0a48e8f815DC0cda2CDb275e0F163A480b']
+      }"
     >
-      <template #chainColumnHeader>
+      <template v-slot:chainColumnHeader>
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/components/contextual/pages/vebal/LMVoting/GaugesTable.vue:179]: This was suprisingly not working! hahahaha

[the remark this answers is not in the document]
Comment (1saf) [reply src/components/contextual/pages/vebal/LMVoting/GaugesTable.vue:179]: for me anyway, locally

[the remark this answers is not in the document]
Comment (garethfuller) [reply src/components/contextual/pages/vebal/LMVoting/GaugesTable.vue:179]: Are you sure that's what the problem was?

**Pull request merged**